From ac5d0ee133cd921c7dd159fd460027a77c75b36a Mon Sep 17 00:00:00 2001
From: Mark Stapp <mjs@cisco.com>
Date: Mon, 21 Jul 2025 16:11:11 -0400
Subject: [PATCH 2/5] lib,zebra: add resolved-via id to show output

Add the resolved-via value to show output (vty and json
forms).

Signed-off-by: Mark Stapp <mjs@cisco.com>
---
 lib/nexthop.c     | 3 +++
 zebra/zebra_vty.c | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/lib/nexthop.c b/lib/nexthop.c
index a8b30dc9b4..2b49eaf31a 100644
--- a/lib/nexthop.c
+++ b/lib/nexthop.c
@@ -1483,6 +1483,9 @@ void nexthop_json_helper(json_object *json_nexthop, const struct nexthop *nextho
 			}
 		}
 	}
+
+	if (nexthop->resolved_via > 0)
+		json_object_int_add(json_nexthop, "resolvedVia", nexthop->resolved_via);
 }
 
 /*
diff --git a/zebra/zebra_vty.c b/zebra/zebra_vty.c
index eff08e2818..674d800c16 100644
--- a/zebra/zebra_vty.c
+++ b/zebra/zebra_vty.c
@@ -325,6 +325,9 @@ static void show_nexthop_detail_helper(struct vty *vty,
 		for (i = 1; i < nexthop->backup_num; i++)
 			vty_out(vty, ",%d", nexthop->backup_idx[i]);
 	}
+
+	if (nexthop->resolved_via > 0)
+		vty_out(vty, ", res via %u", nexthop->resolved_via);
 }
 
 static void zebra_show_ip_route_opaque(struct vty *vty, struct route_entry *re,
-- 
2.50.1 (Apple Git-155)

